crates/amalthea/src/socket/socket.rs

```diff
@@ -133,6 +133,15 @@ impl Socket {
             }
         }
 
+        // 10 second timeout on `recv()` operations (for testing for now)
+        // TODO: Expose `IS_TESTING` in its own lightweight crate?
+        if let Err(error) = socket.set_rcvtimeo(10000) {
+            return Err(Error::CreateSocketFailed(name, error));
+        }
+        if let Err(error) = socket.set_sndtimeo(10000) {
+            return Err(Error::CreateSocketFailed(name, error));
+        }
```

Comment:
TODO: Remove or hide behind test flag

The goal is to avoid an infinite hang on send/recv issues during kernel tests